modules/database/src/adapters/sequelize-adapter/utils/schema.ts

```diff
@@ -136,12 +137,23 @@ export async function getTransactionAndParsedQuery(
   } else {
     parsedQuery = query;
   }
-  if (parsedQuery.hasOwnProperty('$set')) {
-    parsedQuery = parsedQuery['$set'];
+  if (method === 'PUT') {
+    const allFields = Object.keys(parsedQuery);
+    for (const field of Object.keys(parsedQuery)) {
+      if (allFields.indexOf(field) === -1) {
+        parsedQuery[field] = null;
+      }
+    }
+    parsedQuery = { $set: parsedQuery };
+  } else if (method === 'PATCH') {
+    if (parsedQuery.hasOwnProperty('$set')) {
```

Comment:
you need to check if the property is missing not if it is already there. If method is PATCH and parsedQuery has $set, then nothing is needed. If method is PATCH and $set is missing you add it. Also if method is PUT but $set exists then we just do patch. $set shouldn't be added on PUT